src/components/user-form/AddUserForm.js

```diff
@@ -2,102 +2,128 @@ import React, { Component, createRef } from 'react';
 
-    const pattern = /\d+/;
-
-    if (pattern.test(firstName) || pattern.test(lastName))  {
-      this.setState({
-        warning: 'please don\'t put numbers '
-      });
-      return;
     }
 
-    console.log(firstName, lastName, email);
-    // todo 1: использовать функцию onUserAdd для создания нового пользователя здесь
-    //  у этого объекта должны быть проперти first_name, last_name, address, email
-
-    // todo 3: зачистить форму, вызвав функцию onReset
-  };
-
-  // todo 3: добавить функцию onReset, которая будет зачищать поля формы
-  //  обратите внимание, что тут нет стейта и мы обращаемся к инпутам с помощью ref
-
-  focusInput = () => {
-    console.log('focused');
-    this.lastNameRef.current.focus()
-  };
-
-  render() {
-    return (
-      <form onSubmit={this.onSubmit}>
-        {!!this.state.warning && <div>{this.state.warning}</div>}
-        <div className="form-group">
-          <label htmlFor="firstNameInput">First name:</label>
-          <input
-            ref={this.firstNameRef}
-            type="text"
-            className="form-control"
-            id="firstNameInput"
-            placeholder="Example input"
-          />
-        </div>
-
-        <div className="form-group">
-          <label htmlFor="lastNameInput">Last name:</label>
-          <input
-            ref={this.lastNameRef}
-            type="text"
-            className="form-control"
-            id="lastNameInput"
-            placeholder="Example input"
-          />
-        </div>
-
-        <div className="form-group">
-          <label htmlFor="lastNameInput">Email:</label>
-          <input
-            ref={this.emailNameRef}
-            type="text"
-            className="form-control"
-            id="lastNameInput"
-            placeholder="Example input"
-          />
-        </div>
-
-        {/*
+    onSubmit = (e) => {
+        // todo 1: достать из props функцию добавления пользователя onUserAdd
+        const { onAddUser } = this.props;
+        e.preventDefault();
+
+        console.log(this.firstNameRef.current.value);
+        console.log(this.lastNameRef.current.value);
+
+        const first_name = this.firstNameRef.current.value;
+        const last_name = this.lastNameRef.current.value;
+        const email = this.emailNameRef.current.value;
+        const address = this.addressNameRef.current.value;
+
+        const pattern = /\d+/;
+
+        if (pattern.test(first_name) || pattern.test(last_name)) {
+            this.setState({
+                warning: 'please don\'t put numbers '
+            });
+            return;
+        }
+
+        console.log(first_name, last_name, email, address);
+            onAddUser({ first_name, last_name, email, address })
+        // todo 1: использовать функцию onUserAdd для создания нового пользователя здесь
+        //  у этого объекта должны быть проперти first_name, last_name, address, email
+
+        // todo 3: зачистить форму, вызвав функцию onReset
```

Comment:
onReset is not called here
the idea was to reset filled inputs after successful adding user